### PR TITLE
Reviewed the custom authenticator docs

### PR DESCRIPTION
**Pull request by himeshsiriwardana**

### CODE REVIEW

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 1]
```diff
@@ -1,161 +1,160 @@
 # Writing a Custom Local Authenticator
 
-The default WSO2 Identity Server authenticator is the basic authenticator. It is a local authenticator that authenticates the end users using a connected user store and the provided user name and password. When a user needs to be authenticated with the user store and authorized based on the assigned user role, you can write a custom local authenticator. Let's try out a sample scenario on this.
+A basic authenticator is available in the WSO2 Identity Server by default. A basic authenticator authenticates end users using a connected userstore and the provided username and password.
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:3]:
```suggestion
A basic authenticator is available in the WSO2 Identity Server by default. A basic authenticator authenticates end users using a connected user store and the provided username and password.
```

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 2]
```diff
@@ -1,161 +1,160 @@
...
+
+With the WSO2 Identity Server, you can write your own local authenticator to define various authentication logic.
 
 ### Sample scenario
-You have an app called playground that is used for importing/exporting photos. The playground uses OpenID Connect to let the users log into it. The playground only allows the users who are assigned with the photoSharingRole user role to log in to it. (The users with other user roles are not allowed to log in to the playground application)
+You have the **playground2** app that you want users to log in with their telephone numbers instead of usernames. Once a user enters a telephone number, your authentication logic should identifiy the user and validate the user's credentials.
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:8]: Is it the **playground2** app or the **pickup-dispatch** app?
@himeshsiriwardana

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 3]
```diff
@@ -179,227 +176,269 @@ Let's begin.
...
+
+                if (log.isDebugEnabled()) {
+                    log.debug("UnSetting the Realm Service");
+                }
+                SampleLocalAuthenticatorServiceComponent.realmService = null;
             }
+        }
         ```
 
 4.  Build the project using maven. (Click to see the sample project
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:423]:
```
4. Build the project using maven. 

      !!! Info
          See the [sample project](https://github.com/wso2/samples-is/tree/master/authenticators/components/org.wso2.carbon.identity.sample.local.authenticator)) for details.
```

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 4]
```diff
@@ -179,227 +176,269 @@ Let's begin.
...
-    [here](../../assets/attachments/org.wso2.custom.authenticator.local.zip)).
+    [here](https://github.com/wso2/samples-is/tree/master/authenticators/components/org.wso2.carbon.identity.sample.local.authenticator)).
 
-5.  Copy the .jar file
+5.  Copy and paste the .jar file
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:426]:
```
5. Copy the .jar file (`org.wso2.custom.authenticator.local-1.0.0.jar `) from the **target** folder in your project directory and copy it to the `<IS_HOME>/repository/components/dropins ` folder.
```

Is this .jar file name correct?

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 5]
```diff
@@ -410,20 +449,18 @@ Let's begin.
 
 13. Under **Local & Outbound Authentication Configuration**, select **Local
     authentication** check box.
-    ![Local authentication check box](../assets/img/using-wso2-identity-server/local-authentication-check-box.jpeg) 
+    ![Local authentication check box](../assets/img/using-wso2-identity-server/local-authentication-check-box.png) 
 
     !!! note
-        On the corresponding drop-down list, you can see, `BasicCustom` which is the display name of the custom authenticator that was written. From the above step, you can make sure your custom authenticator is there and ready for use.    
+        On the corresponding drop-down list, you can see, `sample-local-authenticator` which is the display name of the custom authenticator that was written. Thus, you can make sure your custom authenticator is there and ready for use.
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:455]:
```suggestion
        On the corresponding drop-down list, you can see `sample-local-authenticator`, which is the display name of the custom authenticator that was written. Thus, you can make sure your custom authenticator is there and ready for use.    
```

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 6]
```diff
@@ -410,20 +449,18 @@ Let's begin.
...
+    ![Local authentication check box](../assets/img/using-wso2-identity-server/local-authentication-check-box.png) 
 
     !!! note
-        On the corresponding drop-down list, you can see, `BasicCustom` which is the display name of the custom authenticator that was written. From the above step, you can make sure your custom authenticator is there and ready for use.    
+        On the corresponding drop-down list, you can see, `sample-local-authenticator` which is the display name of the custom authenticator that was written. Thus, you can make sure your custom authenticator is there and ready for use.    
 
-14. Visit the playground app and provide **Client ID** of the registered
-    playground app and give the **Scope** as `openid` to make sure it is
-    in the OpenID Connect flow. Click **Authorize**.  
+14. Visit the playground2 app and provide the **Client ID** of the registered
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:457]: playground2 app?

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 7]
```diff
@@ -1,161 +1,160 @@
 # Writing a Custom Local Authenticator
 
-The default WSO2 Identity Server authenticator is the basic authenticator. It is a local authenticator that authenticates the end users using a connected user store and the provided user name and password. When a user needs to be authenticated with the user store and authorized based on the assigned user role, you can write a custom local authenticator. Let's try out a sample scenario on this.
```
Comment (pamodaaw) [inline en/docs/develop/writing-a-custom-local-authenticator.md:3]:
```
The default WSO2 Identity Server authenticator is the basic authenticator. It is a local authenticator that authenticates the end users using a connected user store and the provided user name and password.
```
this more meaningful

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 8]
```diff
@@ -179,227 +176,269 @@ Let's begin.
...
-        }
-        
-        @Override
-        public String getName() {
-            return BasicCustomAuthenticatorConstants.AUTHENTICATOR_NAME;
-        }
         }
         ```    
 
+        !!! note
```
Comment (pamodaaw) [inline en/docs/develop/writing-a-custom-local-authenticator.md:320]:
```suggestion
        !!! Note    
```

File: en/docs/develop/writing-a-custom-local-authenticator.md (hunk 4 above)
Comment (pamodaaw) [reply en/docs/develop/writing-a-custom-local-authenticator.md:426]: `org.wso2.carbon.auth.local.sample-1.0.0.jar`

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 9]
```diff
@@ -179,227 +176,269 @@ Let's begin.
...
 6.  Start WSO2 Identity Server.
 
-7.  Sign in to the Management Console.
+7.  Sign in to the **Management Console**.
 
-8.  Create a new user with the name `Larry`.
+8.  Create a new user named `Larry`.
 
-9.  Create a new user role called  `photoSharingRole` and assign it to the newly created user.
+9.  Go to the user's profile and update the telephone number.
```
Comment (pamodaaw) [inline en/docs/develop/writing-a-custom-local-authenticator.md:438]: Mention that it's Lary's profile

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 10]
```diff
@@ -179,227 +176,269 @@ Let's begin.
...
     `           <IS_HOME>/repository/components/dropins          `
     folder.
 
 6.  Start WSO2 Identity Server.
 
-7.  Sign in to the Management Console.
+7.  Sign in to the **Management Console**.
 
-8.  Create a new user with the name `Larry`.
+8.  Create a new user named `Larry`.
```
Comment (pamodaaw) [inline en/docs/develop/writing-a-custom-local-authenticator.md:436]: do we need to link docs on how to create the user and update the profile?

File: en/docs/develop/writing-a-custom-local-authenticator.md (hunk 5 above)
Comment (pamodaaw) [reply en/docs/develop/writing-a-custom-local-authenticator.md]: "custom authenticator is deployed correctly and ready for use. "

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 11]
```diff
@@ -410,20 +449,18 @@ Let's begin.
...
+14. Visit the playground2 app and provide the **Client ID** of the registered
+    app and give the **Scope** as `openid` to make sure it is
+    using the OpenID Connect flow. Click **Authorize**.  
     ![Playground app](../assets/img/using-wso2-identity-server/playground-app.jpeg)
     Note that you will be directed to WSO2 Identity Server login page.
 
-15. Provide the username and password of the user `Larry` who is in the
-    `photoSharingRole`. You are prompted to approve the app and logged
-    in.
+15. Provide the telephone number as the user identifier and enter the password of the user `Larry`. You are prompted to approve the app and log in.
```
Comment (pamodaaw) [inline en/docs/develop/writing-a-custom-local-authenticator.md:463]: for the first time, the consent page also can be there

File: en/docs/develop/writing-a-custom-local-authenticator.md (hunk 10 above)
Comment (nilminiwso2) [reply en/docs/develop/writing-a-custom-local-authenticator.md:436]: yes +1

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 12]
```diff
@@ -410,20 +449,18 @@ Let's begin.
 
 13. Under **Local & Outbound Authentication Configuration**, select **Local
     authentication** check box.
```
[the remark this answers is not in the document]
Comment (nilminiwso2) [reply en/docs/develop/writing-a-custom-local-authenticator.md:451]:
```
13. Under **Local & Outbound Authentication Configuration**, select **Local
    authentication**.
```
It should be the above

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 13]
```diff
@@ -179,251 +176,291 @@ Let's begin.
...
     !!! note
-        On the corresponding drop-down list, you can see, `BasicCustom` which is the display name of the custom authenticator that was written. From the above step, you can make sure your custom authenticator is there and ready for use.    
+        On the corresponding drop-down list, you can see, `sample-local-authenticator` which is the display name of the custom authenticator that was written. Thus, you can make sure your custom authenticator is there and ready for use.    
 
-14. Visit the playground app and provide **Client ID** of the registered
-    playground app and give the **Scope** as `openid` to make sure it is
-    in the OpenID Connect flow. Click **Authorize**.  
-    ![Playground app](../assets/img/using-wso2-identity-server/playground-app.jpeg)
-    Note that you will be directed to WSO2 Identity Server login page.
+14. Configure the pickup-dispatch app with the appropriate **Client ID**. Learn how to [deploy the pickup-dispatch webapp](https://is.docs.wso2.com/en/5.11.0/learn/deploying-the-sample-app/#deploying-the-pickup-dispatch-webapp)
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:457]: break this into two.
CLient Id and Client secret

File: en/docs/develop/writing-a-custom-local-authenticator.md [hunk 14]
```diff
@@ -179,251 +176,291 @@ Let's begin.
...
     `           <IS_HOME>/repository/components/dropins          `
     folder.
 
 6.  Start WSO2 Identity Server.
 
-7.  Sign in to the Management Console.
+7.  Sign in to the **Management Console**.
 
-8.  Create a new user with the name `Larry`.
+8.  Create a new user named `Larry`. Learn how to [Create a user](https://is.docs.wso2.com/en/latest/learn/adding-users-and-roles/#create-a-user)
```
Comment (nilminiwso2) [inline en/docs/develop/writing-a-custom-local-authenticator.md:435]: link has to be the relative link.

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**